neurodamus/report.py

```diff
@@ -48,43 +53,166 @@ def get_section_index(cell, section):
 
 
 class Report:
+    """Abstract base class for handling simulation reports in NEURON.
+
+    Provides methods for parsing report parameters, handling variables, and defining the structure
+    required by subclasses to append specific data (e.g., compartments or currents).
+    """
+
     INTRINSIC_CURRENTS = {"i_membrane", "i_membrane_", "ina", "ica", "ik", "i_pas", "i_cap"}
     CURRENT_INJECTING_PROCESSES = {"SEClamp", "IClamp"}
 
+    class ScalingMode(IntEnum):
+        """Enum to define scaling modes used in report generation.
+
+        SCALING_NONE: No scaling.
+        SCALING_AREA: Scale by membrane area.
+        SCALING_ELECTRODE: Custom/electrode-based scaling (fallback/default).
+        """
+
+        SCALING_NONE = 0
+        SCALING_AREA = 1
+        SCALING_ELECTRODE = 2
+
+        @classmethod
+        def from_option(cls, option):
+            """Map user-provided scaling option to a ScalingMode enum.
+
+            :param option: User-specified string or None.
+            :return: Corresponding ScalingMode.
+            """
+            mapping = {
+                None: cls.SCALING_AREA,
+                "Area": cls.SCALING_AREA,
+                "None": cls.SCALING_NONE,
+            }
+            return mapping.get(option, cls.SCALING_ELECTRODE)
+
     def __init__(
         self,
-        report_name,
-        report_type,
-        variable_name,
-        unit,
-        _format,
-        dt,
-        start_time,
-        end_time,
-        output_dir,
-        scaling_option=None,
-        use_coreneuron=False,
+        params,
+        use_coreneuron,
     ):
-        self.variable_name = variable_name
-        self.report_dt = dt
-        self.scaling_mode = self.determine_scaling_mode(scaling_option)
+        """Initialize a Report object with simulation parameters.
+
+        :param params: Object containing report configuration (e.g., name, dt, unit).
+        :param use_coreneuron: Boolean indicating if CoreNEURON is enabled.
+        """
+        if type(self) is Report:
+            raise TypeError("Report is an abstract base class and cannot be instantiated directly.")
+
+        self.variable_name = params.report_on
+        self.report_dt = params.dt
+        self.scaling_mode = self.ScalingMode.from_option(params.scaling)
         self.use_coreneuron = use_coreneuron
 
         self.alu_list = []
         self.report = Nd.SonataReport(
-            0.5, report_name, output_dir, start_time, end_time, dt, unit, "compartment"
+            0.5,
+            params.name,
+            params.output_dir,
+            params.start,
+            params.end,
+            params.dt,
+            params.unit,
+            "compartment",
         )
         Nd.BBSaveState().ignore(self.report)
 
+    def append_gid_section(
```

Comment:
Can we rename the function to `register_section_report`?  Like what is in coreneuron
https://github.com/neuronsimulator/nrn/blob/20e6a13dded2f1923c61f1d6e1b29f7bbf586398/src/coreneuron/io/reports/report_handler.cpp#L76

Reply:
ah I did not know. My version is a little clearer imho but the neurodamus one is still fine. I will do it

Reply:
done